common/src/main/java/earth/terrarium/prometheus/common/commands/utilities/RtpCommand.java

```diff
@@ -74,8 +74,11 @@ public static BlockPos tp(BlockPos location, ServerPlayer player, int distance,
         final int min = distance / 4;
         final int max = distance - min;
 
-        int x = min + player.getRandom().nextInt(-max, max) + location.getX();
-        int z = min + player.getRandom().nextInt(-max, max) + location.getZ();
+        int randRangeX = min + player.getRandom().nextInt(max);
+        int x = location.getX() + (randRangeX * ((Math.random() < 0.5) ? -1 : 1));
```

Comment:
math.random should not be used for this player.getRandom should be used due to the performance difference

Reply:
Will do; bit surprised that would be significant in this situation (a player asking to rtp). TIL, I guess.